gobblin-service/src/main/java/org/apache/gobblin/service/modules/flow/MultiHopFlowCompiler.java

```diff
@@ -281,6 +281,12 @@ public Dag<JobExecutionPlan> compileFlow(Spec spec) {
     Instrumented.markMeter(flowCompilationSuccessFulMeter);
     Instrumented.updateTimer(flowCompilationTimer, System.nanoTime() - startTime, TimeUnit.NANOSECONDS);
 
+    if (Boolean.parseBoolean(flowSpec.getConfigAsProperties().getProperty(ServiceConfigKeys.GOBBLIN_SERVICE_ADHOC_FLOW))) {
+      // todo : we should probably set it on all of the start nodes
```

Comment:
I don't think it should be set for all the nodes because multihop flows would break. When each hop is completed it would decrement the quota and re-increment the quota for the next hop. If it's set for all nodes then this scenario would need to be handled

Reply:
comment is not about all the nodes, only about the start nodes.